telemetry/software/telemetry/library.py

```diff
@@ -0,0 +1,116 @@
+"""
+objective - grab data from serial, parse, pass into redis
+timeseries database in realtime
+"""
+
+import serial
+import time
+from redis import Redis
+from redistimeseries.client import Client
+
+
+# Data stream parent class with main pass_data() method call
+# 2-3 data stream children - serial stream, logfile stream, custom pass_data() methods
+class DataStream(object):
+    def __init__(self, data_channels):
+        # self.start_stream = True
+        self.data_channels = data_channels
+
+    # def close_stream(self):
+    #     self.start_stream = False
+
+    def return_data_channels(self):
+        return self.data_channels
+
+
+class ArduinoSerialIn(DataStream):
+    """
+    Class to represent arduino serial datastream object in for testing purposes
+    """
+
+    def __init__(
+        self, baudrate=9600, port_name="/dev/ttyUSB0", data_channels=["arduino_data"]
```

Comment:
9600 baud is not going to cut it. Luke mentioned that he's trying to get the RF module at 500kbps, 9600bps is going to be your bottleneck here. What are the tradeoffs to increasing the baudrate of the UART comms?

Reply:
I dont have an answer to your question about tradeoffs @wsh32 but I'm using a baudrate of 115200 (the upper end of "standard") on the HITL project which I've never run into limitations with. That said, if we are actually trying to stream 500K bps (I assume 1 fully saturated CAN bus?), then even 115200 will be limiting without aggressive compression.

Reply:
Other "standard" baudrates are 19200, 38400, 57600, 115200, and 256000.

> The higher a baud rate goes, the faster data is sent/received, but there are limits to how fast data can be transferred. You usually won't see speeds exceeding 115200 - that's fast for most microcontrollers. Get too high, and you'll begin to see errors on the receiving end, as clocks and sampling periods just can't keep up.

[Sparkfun](https://learn.sparkfun.com/tutorials/serial-communication/all)

Reply:
Also, I need to put together a presentation on udev (and maybe config files in general) which will save us from having to guess the port name every time.